test/jdk/java/awt/Frame/Iconify/IconifiedToFront.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, JetBrains s.r.o.. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @key headful
+ * @bug 8326497
+ * @summary Verifies that an iconified window is restored with Window.toFront()
+ * @library /test/lib
+ * @run main IconifiedToFront
+ */
+
+import java.awt.Color;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Label;
+import java.awt.Robot;
+import java.awt.Toolkit;
+
+public class IconifiedToFront {
+    private static final int PAUSE_MS = 1500;
```

Comment:
1.5 sec is minimum delay, can we reduced it ?

Reply:
Not unless there's another idea on how to measure tests' success. 1.5 seconds is about the time my MacOS needs to complete an iconify/de-iconfiy process.

Reply:
Did u try after adding `waitForIdle` ?

Reply:
You were right @TejeshR13. `waitForIdle` helps and I reverted the time back to 500ms.